web/src/lib/components/asset-viewer/photo-viewer.svelte

```diff
@@ -1,98 +1,92 @@
-      maxZoom: 10,
-      wheelZoomRatio: 0.2,
-    });
+  let assetFileUrl: string = '';
+  let imageLoaded: boolean = false;
+  let imageError: boolean = false;
+  // when true, will force loading of the original image
+  let forceLoadOriginal: boolean = false;
+
+  $: {
+    preload({ preloadAssets, loadOriginal: loadOriginalByDefault });
   }
 
-  onMount(async () => {
-    // Import hack :( see https://github.com/vadimkorr/svelte-carousel/issues/27#issuecomment-851022295
-    // TODO: Move to regular import once the package correctly supports ESM.
-    const module = await import('copy-image-clipboard');
-    copyImageToClipboard = module.copyImageToClipboard;
-    canCopyImagesToClipboard = module.canCopyImagesToClipboard;
+  $: imageLoaderUrl = load(asset.id, loadOriginalByDefault || forceLoadOriginal, false, asset.checksum);
+
+  photoZoomState.set({
+    currentRotation: 0,
+    currentZoom: 1,
+    enable: true,
+    currentPositionX: 0,
+    currentPositionY: 0,
   });
 
-  $: void loadAssetData({ loadOriginal: loadOriginalByDefault, checksum: asset.checksum });
+  $zoomed = false;
+
+  const unsubscribeZoomState = photoZoomState.subscribe((state) => {
+    forceLoadOriginal = forceLoadOriginal || (state.currentZoom > 1 && isWebCompatibleImage(asset)) ? true : false;
+    haveFadeTransition = false;
+  });
 
   onDestroy(() => {
     $boundingBoxesArray = [];
-    abortController?.abort();
+    unsubscribeZoomState();
   });
 
-  const loadAssetData = async ({ loadOriginal, checksum }: { loadOriginal: boolean; checksum: string }) => {
-    try {
-      abortController?.abort();
-      abortController = new AbortController();
-
-      // TODO: Use sdk once it supports signals
-      const res = await downloadRequest({
-        url: getAssetFileUrl(asset.id, !loadOriginal, false, checksum),
-        signal: abortController.signal,
-      });
-
-      assetData = window.URL.createObjectURL(res.data);
-      imageLoaded = true;
-
-      if (!preloadAssets) {
-        return;
+  const preload = ({
+    preloadAssets,
+    loadOriginal,
+  }: {
+    preloadAssets: AssetResponseDto[] | null;
+    loadOriginal: boolean;
+  }) => {
+    for (const preloadAsset of preloadAssets || []) {
+      if (preloadAsset.type === AssetTypeEnum.Image) {
+        let img = new Image();
+        img.src = getAssetFileUrl(preloadAsset.id, !loadOriginal, false, preloadAsset.checksum);
       }
-
-      for (const preloadAsset of preloadAssets) {
-        if (preloadAsset.type === AssetTypeEnum.Image) {
-          await downloadRequest({
-            url: getAssetFileUrl(preloadAsset.id, !loadOriginal, false),
-            signal: abortController.signal,
-          });
-        }
-      }
-    } catch {
-      imageLoaded = false;
     }
   };
 
-  const doCopy = async () => {
+  const load = (assetId: string, isOriginal: boolean, isThumb: boolean, checksum: string) => {
+    return getAssetFileUrl(assetId, !isOriginal, isThumb, checksum);
```

Comment:
This function inverts `isOriginal`, but in other places `getAssetFileUrl` is used directly. That's a bit confusing and I would use only one of both.

Reply:
This is going to be a conflict with #9831 anyways, so it will probably have to be rewritten in a cleaner manner.

Reply:
this has been completely rewritten, and much better now